server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/http/MailboxesProvisioner.scala

```diff
@@ -0,0 +1,81 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.http
+
+import java.util.Optional
+
+import javax.inject.Inject
+import org.apache.james.core.Username
+import org.apache.james.mailbox.exception.{MailboxException, MailboxExistsException}
+import org.apache.james.mailbox.model.{MailboxId, MailboxPath}
+import org.apache.james.mailbox.{DefaultMailboxes, MailboxManager, MailboxSession, SubscriptionManager}
+import org.apache.james.metrics.api.MetricFactory
+import org.apache.james.util.FunctionalUtils.negate
+import org.slf4j.{Logger, LoggerFactory}
+import reactor.core.scala.publisher.{SFlux, SMono}
+import reactor.core.scheduler.Schedulers
+
+class MailboxesProvisioner @Inject() (mailboxManager: MailboxManager,
+                                      subscriptionManager: SubscriptionManager,
+                                      metricFactory: MetricFactory) {
+  private val LOGGER: Logger = LoggerFactory.getLogger(classOf[MailboxesProvisioner])
+
+  def createMailboxesIfNeeded(session: MailboxSession): SMono[Void] =
+    metricFactory.decorateSupplierWithTimerMetric("JMAP-mailboxes-provisioning", () => {
+      val username: Username = session.getUser
+      createDefaultMailboxes(username)
```

Comment:
createDefaultMailboxes(session.getUser) and get rid of the {...} would be shorter?

Reply:
Didn't finish to work on this part, it's a WIP. But thanks for the feedback :)